.github/workflows/golangci-lint.yml

```diff
@@ -18,6 +18,9 @@ jobs:
   golangci:
     name: golangci-lint
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        mode: [no-deprecations, no-failure]
```

Comment:
Consider if we could simplify by only going with the no-deprecations mode. If we want to see the deprecations we can run golangci-lint manually.

Reply:
I had that originally, but I didn't want to lose track of the deprecations by requiring additional local steps.  It's cheap enough in CI, and I like being able to keep an eye on them.